Gordon360/ApiControllers/HousingController.cs

```diff
@@ -188,16 +192,16 @@ public IHttpActionResult SaveApplication([FromBody] ApartmentAppViewModel apartm
         /// </summary>
         /// <returns>Returns the application ID number if all the queries succeeded</returns>
         [HttpPut]
-        [Route("apartment/save")]
-        public IHttpActionResult EditApplication([FromBody] ApartmentAppViewModel apartmentAppDetails)
+        [Route("apartment/applications/{applicationID}")]
+        public IHttpActionResult EditApplication(int applicationID, [FromBody] ApartmentApplicationViewModel applicationDetails)
```

Comment:
It is not redundant, it is to follow the proper RESTful design. The routes should signify which collection you are interaction with, and then the inline parameter tells it which "element" of that collection (in a abstract sense). So in this case, you want to `HttpPut` to update the item corresponding to `{applicationID}` from the collection of `/applications/`,  and then `/applications` is a sub-set of `apartment` (It is essentially to reflect how you would think of the path to the thing from the perspective of it all being a nested tree structure,)

You are specifically **not** supposed to use the naming of the route to describe what you are _doing_ to the data, 
rather, the route tells the API _what_ or _which_ **data** you want to interact with, and the HTTP get, post, put, etc. is how you communicate what **action** or **operation** you want to do with that data. (There are much more clear, concise, and/or well-written explanations in real books or documentation; this is my attempt to paraphrase based on how I like to think of it)

Reply:
So to answer your first question more directly, the extra parameter is **not** there to distinguish between the Put and the Post.
The extra parameter is there to distinguish which application you are trying to edit, because the proper RESTful design is to provide that information in the URL of the route like I did here, not by lumping it into the JSON with all the other data. It's like is a key-value pair, expect more like key-[many values], so you put the key in the route URL rather than in the JSON so that it is clear and obvious which stuff is the key/ID versus which stuff is the data associated with that key/ID

Reply:
Okay, that's good then. Thank you for the mini-lesson there :)